### ISSUE TITLE
Add session windows support 

### ISSUE THREAD

**Issue by qoega** [ClickHouse/ClickHouse#45311]
**Use case**

When you have stream of events for a user it is useful to split the whole data to a logical sessions based on user activity. Simplest way to to do is to consider session as ended after some period of inactivity i.e `session_timeout`. For some cases `maximum_session_time` is used to lower session size and split them even without user idle period.

We want to split data into separate sessions and perform operations with separate sessions.
It can be 
1) Trivially assign session id to corresponding rows (it can be treated as dictGet or join with sessions).
2) Aggregate over a session window (trivial group by after 1. is supported)
3) Support WINDOW functions inside session window like `ROW_NUMBER()`

**Describe the solution you'd like**

1) Choose SQL syntax for session windows in ClickHouse.
2) Support session windows.
3) [Part 2. Should be done as a separate issue.] Make session windows work with Materialized Views. If we perform the same operation with new data only it will 
  - not join previous interval i.e. result will depend on inserts order.
  - If we find a way to join them we may need to replace values of previous MV run: we can use interval start as window function id to support growing intervals. 
  3. is quite complex and requires either `global time` tracking that allows to process data treating that all data till some point in time is already received and we may perform operation on previous data. Or we need to add some stateful processing of MVs that allows to get previous operation state before processing new insert.

**Additional context**

- [Azure Stream analytics](https://learn.microsoft.com/en-us/azure/stream-analytics/stream-analytics-window-functions)
- [Apache Spark](https://spark.apache.org/docs/latest/structured-streaming-programming-guide.html) or [this](https://towardsdatascience.com/spark-3-2-session-windowing-feature-for-streaming-data-e404d92e267)
- [kSQL](https://docs.ksqldb.io/en/latest/concepts/time-and-windows-in-ksqldb-queries/#session-window)
- [Apache Flink](https://nightlies.apache.org/flink/flink-docs-master/docs/dev/datastream/operators/windows/#session-windows)

**Comment by UnamedRus** [comment]
AFAIK, there is SQL standard MATCH RECOGNIZE for that

https://blogs.oracle.com/sql/post/ogb-appreciation-day-sql-pattern-matching-match-recognize-use-cases#batching
https://docs.snowflake.com/en/sql-reference/constructs/match_recognize.html

**Comment by DG-Wangtao** [comment]
Hi, @UnamedRus :
I have tried MATCH_RECOGNIZE in Clickhouse but I got some errors. I found an issue #37577 you created, is that mean the `MATCH_RECOGNIZE` is not supported now in click house? Are there any alternatives in Clickhouse?
Thank you.

**Comment by jovezhong** [comment]
I noticed a PR sits there more than 1 year.

If you are blocked on this, maybe you can try https://github.com/timeplus-io/proton You can read from ClickHouse or Kafka, apply the session window aggregation, then write back to ClickHouse or Kafka

```sql
-- https://docs.timeplus.com/functions_for_streaming#session
SELECT window_start, count(), avg(col1), p99(col2)
FROM session(source, 5s) --5 sec as the idle time to break session or set a session start/end condition
GROUP BY window_start
```
